### PR TITLE
network: fix BridgeVLAN issue

### LINKED ISSUE #12958: [networkd] Adding VLAN 4090 to VLAN-aware bridge is not working
**systemd version the issue has been seen with**
> 241

**Used distribution**
 > Fedora 30

**Expected behaviour you didn't see**
 > Expected to see the VLAN 4090 applied to the created bridge, able to see via `bridge vlan` command. 

**Unexpected behaviour you saw**
 > The VLAN wasn't applied to bridge, and systemd-networkd output:
```
dummy0: Could not append VLANs: Invalid argument
dummy0: Failed to assign VLANs to bridge port: Invalid argument                                                                                                            
dummy0: Could not set bridge vlan: Invalid argument
[...]
swbridge: Could not append VLANs: Invalid argument
swbridge: Failed to assign VLANs to bridge port: Invalid argument
swbridge: Could not set bridge vlan: Invalid argument
```

**Steps to reproduce the problem**
ip link add type dummy
Networkd files:

20-dummy0.network
```
[Match]
Name=dummy0
    
[Link]
RequiredForOnline=no
    
[Network]
Bridge=swbridge

[BridgeVLAN]
VLAN=4090
```
30-swbridge.netdev
``` 
[NetDev]
Name=swbridge
Kind=bridge
    
[Bridge]
VLANFiltering=1
ForwardDelaySec=2
DefaultPVID=none
STP=true
```
30-swbridge.network
```
[Match]
Name=swbridge

[Link]
RequiredForOnline=no

[BridgeVLAN]
VLAN=4090
```

### PR DESCRIPTION
**Pull request by yuwata**
Closes #12958.

### CODE REVIEW

**Comment by evverx** [comment]
Looks like UBSan is complaining
```
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]: ../src/network/networkd-brvlan.c:22:27: runtime error: left shift of 1 by 31 places cannot be represented in type 'int'
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #0 0x55973b49d6b9 in is_bit_set ../src/network/networkd-brvlan.c:22
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #1 0x55973b49de1e in append_vlan_info_data ../src/network/networkd-brvlan.c:78
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #2 0x55973b49f4d2 in br_vlan_configure ../src/network/networkd-brvlan.c:187
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #3 0x55973b4f18f8 in link_set_bridge_vlan ../src/network/networkd-link.c:1170
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #4 0x55973b4ff46c in link_joined ../src/network/networkd-link.c:1995
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #5 0x55973b502c64 in link_enter_join_netdev ../src/network/networkd-link.c:2139
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #6 0x55973b508314 in link_configure_after_setting_mtu ../src/network/networkd-link.c:2578
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #7 0x55973b507f44 in link_configure ../src/network/networkd-link.c:2559
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #8 0x55973b50ae08 in link_initialized_and_synced ../src/network/networkd-link.c:2772
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #9 0x55973b50ae96 in link_initialized_handler ../src/network/networkd-link.c:2780
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #10 0x7efe6c508fab in process_reply ../src/libsystemd/sd-netlink/sd-netlink.c:347
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #11 0x7efe6c509dc7 in process_running ../src/libsystemd/sd-netlink/sd-netlink.c:414
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #12 0x7efe6c50a27d in sd_netlink_process ../src/libsystemd/sd-netlink/sd-netlink.c:443
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #13 0x7efe6c50cb03 in time_callback ../src/libsystemd/sd-netlink/sd-netlink.c:725
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #14 0x7efe6c55ec37 in source_dispatch ../src/libsystemd/sd-event/sd-event.c:2835
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #15 0x7efe6c565206 in sd_event_dispatch ../src/libsystemd/sd-event/sd-event.c:3240
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #16 0x7efe6c56619b in sd_event_run ../src/libsystemd/sd-event/sd-event.c:3297
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #17 0x7efe6c5664c8 in sd_event_loop ../src/libsystemd/sd-event/sd-event.c:3319
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #18 0x55973b3c39c8 in run ../src/network/networkd.c:114
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #19 0x55973b3c3bdf in main ../src/network/networkd.c:121
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #20 0x7efe6ac1ace2 in __libc_start_main (/lib64/libc.so.6+0x23ce2)
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]:     #21 0x55973b3c1cbd in _start (/build/build/systemd-networkd+0x2cecbd)
Jul 04 14:26:30 arch.localdomain systemd-networkd[2360]: SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/network/networkd-brvlan.c:22:27 in


 ____________________________________________
/ Found   1 sanitizer errors (  0 ASan,   1  \
| UBSan,   0 MSan). Looks like you need to   |
\ look at the log                            /
 --------------------------------------------
 \
  \
     __
    /  \
    |  |
    @  @
    |  |
    || |/
    || ||
    |\_/|
    \___/
```

File: src/network/networkd-brvlan.c [hunk 1]
```diff
@@ -204,57 +204,15 @@ int br_vlan_configure(Link *link, uint16_t pvid, uint32_t *br_vid_bitmap, uint32
...
-                        return r;
-
-                if (_vid == 0)
-                        return -ERANGE;
-        }
-
-        *vid = _vid;
-        *vid_end = _vid_end;
-        return r;
-}
```
Comment (keszybz) [inline src/network/networkd-brvlan.c:248]: This chunk belongs in the next commit...

**Comment by yuwata** [comment]
Thank you for the comments. I've force-pushed a revised version. I hope now the UBSan issue is fixed.

**Comment by yuwata** [comment]
Yey! All green now.

**Pull request merged**